engine/player/unique_gear.cpp

```diff
@@ -1828,6 +1830,24 @@ void item::infallible_tracking_charm( special_effect_t& effect )
   new dbc_proc_callback_t( effect.item, effect );
 }
 
+void item::witherbarks_branch( special_effect_t& effect )
+{
+  auto data_spell = effect.player->find_spell( 429257 );
+  auto stat_buff  = make_buff<stat_buff_t>( effect.player, "aqueous_dowsing", effect.player->find_spell( 429257 ) )
+    ->add_stat( STAT_MASTERY_RATING, data_spell->effectN( 1 ).average( effect.item ) );
```

Comment:
Looks like you've got the wrong spell id here for the buff, should be `429262`

Also should be named `aqueous_enrichment`

with that fixed, you dont need the `add_stat` function all all, since the value, the stat it gives, etc, will be parsed from the spell data.

doing this, will automatically parse most of the buff effects, such as the 10s duration, the 3 stack cap, and what stat it gives (in case it ever changes in the future). 

final note, dont need to store the data spell, since its just the driver, can call that spell with `effect.driver()` wherever you need to call one of its effects.